plugins/rich-editor/src/scripts/components/EditorEmojiPicker.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * @author Stéphane (slafleche) LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import React from "react";
+import * as PropTypes from "prop-types";
+import Quill from "quill/quill";
+import EditorEmojiMenu from "../components/EditorEmojiMenu";
+import { t } from "@core/utility";
+
+
+export default class EditorEmojiPicker extends React.Component {
+
+    /** @type {number} */
+    static count;
+
+    static propTypes = {
+        quill: PropTypes.instanceOf(Quill).isRequired,
+    };
+
+    /** @type {Quill} */
+    quill;
+
+    /**
+     * @type {Object}
+     * @property {RangeStatic} - The current quill selected text range..
+     */
+    state;
+
+    /**
+     * @inheritDoc
+     */
+    constructor(props) {
+        super(props);
+
+        if (!this.count) {
+            this.count = 1;
+        } else {
+            this.count++;
+        }
+
+        this.editorID = this.count;
+        this.menuID = "emojiMenu-menu-" + this.editorID;
+        this.buttonID = "emojiMenu-button-" + this.editorID;
+        this.menuTitleID = "emojiMenu-title-" + this.editorID;
+
+        // Quill can directly on the class as it won't ever change in a single instance.
+        this.quill = props.quill;
+
+        this.state = {
+            isVisible: false,
+        };
+
+        this.escFunction = this.escFunction.bind(this);
+
+    }
+
+    escFunction(event){
+        if(event.keyCode === 27) {
+            this.closeMenu();
```

Comment:
You're not passing the event through which is why e.preventDefault() isn't working in the closeMenu function. Could these 2 functions be merged together?